public final class c.g.a.e.m extends java.lang.Object implements c.g.a.g.f
{
    public static int a;
    public static int b;
    public final c.g.a.g.s c;

    public void <init>(c.g.a.g.s)
    {
        c.g.a.e.m r0;
        c.g.a.g.s $r1;

        r0 := @this: c.g.a.e.m;

        $r1 := @parameter0: c.g.a.g.s;

        specialinvoke r0.<java.lang.Object: void <init>()>();

        r0.<c.g.a.e.m: c.g.a.g.s c> = $r1;

        return;
    }

    public static c.g.a.g.f a(c.g.a.g.s)
    {
        java.lang.Throwable $r2;
        java.lang.NullPointerException r4;
        byte $b2;
        c.g.a.g.s $r1;
        int $i0, $i1;
        c.g.a.e.m $r0, $r3;

        $r1 := @parameter0: c.g.a.g.s;

        $r3 = new c.g.a.e.m;

        $r0 = $r3;

        specialinvoke $r3.<c.g.a.e.m: void <init>(c.g.a.g.s)>($r1);

        $i0 = <c.g.a.e.m: int a>;

        $i0 = $i0 + 100;

        $i1 = -1;

        $i0 = $i0 + $i1;

        $i1 = $i0 % 128;

        <c.g.a.e.m: int b> = $i1;

        $i0 = $i0 % 2;

        if $i0 != 0 goto label1;

        $b2 = 75;

        goto label2;

     label1:
        $b2 = 91;

     label2:
        if $b2 == 91 goto label5;

     label3:
        r4 = new java.lang.NullPointerException;

        specialinvoke r4.<java.lang.NullPointerException: void <init>(java.lang.String)>("This statement would have triggered an Exception: specialinvoke $u2#9.<java.lang.Object: int hashCode()>()");

        throw r4;

     label4:
        $r2 := @caughtexception;

        throw $r2;

     label5:
        return $r0;

        catch java.lang.Throwable from label3 to label4 with label4;
    }

    public final java.lang.Object a(java.lang.Object)
    {
        java.lang.Throwable $r5;
        java.lang.Exception $r4;
        java.lang.NullPointerException r6;
        com.distil.protection.model.ProtectionException $r1;
        byte $b3;
        c.g.a.g.s $r3;
        int $i0, $i1, $i2;
        java.lang.Object $r2;
        c.g.a.e.m r0;

        r0 := @this: c.g.a.e.m;

        $r2 := @parameter0: java.lang.Object;

        $i0 = <c.g.a.e.m: int b>;

        $i1 = $i0 & 17;

        $i2 = -1;

        $i2 = $i1 ^ $i2;

        $i0 = $i0 | 17;

        $i0 = $i0 & $i2;

        $i1 = $i1 << 1;

        $i1 = $i0 + $i1;

        $i1 = $i1 % 128;

        <c.g.a.e.m: int a> = $i1;

        $r3 = r0.<c.g.a.e.m: c.g.a.g.s c>;

        $r4 = (java.lang.Exception) $r2;

        $r1 = staticinvoke <c.g.a.e.l: com.distil.protection.model.ProtectionException a(c.g.a.g.s,java.lang.Exception)>($r3, $r4);

        $i1 = <c.g.a.e.m: int a>;

        $i0 = -52;

        $i0 = $i1 & $i0;

        $i2 = -1;

        $i2 = $i1 ^ $i2;

        $i2 = $i2 & 51;

        $i0 = $i0 | $i2;

        $i1 = $i1 & 51;

        $i1 = $i1 << 1;

        $i1 = neg $i1;

        $i2 = neg $i1;

        $i1 = $i0 ^ $i2;

        $i0 = $i2 & $i0;

        $i0 = $i0 << 1;

        $i1 = $i1 + $i0;

        $i0 = $i1 % 128;

        <c.g.a.e.m: int b> = $i0;

        $i1 = $i1 % 2;

        if $i1 != 0 goto label1;

        $b3 = 89;

        goto label2;

     label1:
        $b3 = 43;

     label2:
        if $b3 == 43 goto label5;

     label3:
        r6 = new java.lang.NullPointerException;

        specialinvoke r6.<java.lang.NullPointerException: void <init>(java.lang.String)>("This statement would have triggered an Exception: specialinvoke $u0#31.<java.lang.Object: int hashCode()>()");

        throw r6;

     label4:
        $r5 := @caughtexception;

        throw $r5;

     label5:
        return $r1;

        catch java.lang.Throwable from label3 to label4 with label4;
    }

    public static void <clinit>()
    {
        <c.g.a.e.m: int b> = 1;

        <c.g.a.e.m: int a> = 0;

        return;
    }
}
